src/Util/XML.php

```diff
@@ -142,7 +143,7 @@ public static function fromArray(array $array, &$xml, bool $remove_header = fals
 	 */
 	public static function copy(&$source, &$target, string $elementname)
 	{
-		if (count($source->children()) == 0) {
+		if (is_string($source) && count($source->children()) == 0) {
```

Comment:
`is_string()` and `$source->children()` isn't compatible. `$foo->bar()` means object. Can you please double-check this?

Reply:
AFAIK the function is called with both a string and an object.

Reply:
But here `is_string($source) && count($source->children())` doesn't make logically sense. You cannot access a method on a string (first check `is_string()` is true).